sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotNestTypesInArguments.cs

```diff
@@ -62,4 +62,16 @@ public void Method()
             static void DoSomethingElse(ICollection<ICollection<int>> outerCollect) { } // Noncompliant
         }
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/3277
+    public class Repro_3277
+    {
+        public void QuestionMark(List<int?> list)  // Noncompliant FP
+        {
+        }
+
+        public void Explicit(List<Nullable<int>> list) // Noncompliant, there's a nice way out of this
```

Comment:
According to the issue this should also be a FP, right?

Reply:
From my POV the syntax `List<Nullable<int>>` is just as any other generic `Outer<Inner<int>>` that the rule is trying to prevent. And luckily, there exist a very nice way out of this problem.

While the syntax `List<int?>` is clearly a FP due to it's semantic model. Syntactically it's not a nested generic type (even when we know that semantically it is, under the hood).

I'd keep `List<Nullable<int>>` as a True Positive.